transports/httpclient/src/main/java/org/commonjava/maven/galley/transport/htcli/internal/HttpDownload.java

```diff
@@ -55,27 +59,75 @@
 
     private boolean deleteFilesOnPath;
 
+    private final MetricRegistry metricRegistry;
+
+    private final TransportMetricConfig metricConfig;
+
+
     public HttpDownload( final String url, final HttpLocation location, final Transfer target,
                          final Map<Transfer, Long> transferSizes, final EventMetadata eventMetadata, final Http http,
-                         final ObjectMapper mapper )
+                         final ObjectMapper mapper, final MetricRegistry metricRegistry,
+                         final TransportMetricConfig metricConfig )
     {
-        this( url, location, target, transferSizes, eventMetadata, http, mapper, true );
+        this( url, location, target, transferSizes, eventMetadata, http, mapper, true, metricRegistry, metricConfig );
     }
 
     public HttpDownload( final String url, final HttpLocation location, final Transfer target,
                          final Map<Transfer, Long> transferSizes, final EventMetadata eventMetadata, final Http http,
-                         final ObjectMapper mapper, final boolean deleteFilesOnPath )
+                         final ObjectMapper mapper, final boolean deleteFilesOnPath,
+                         final MetricRegistry metricRegistry, final TransportMetricConfig metricConfig )
     {
         super( url, location, http );
         this.target = target;
         this.transferSizes = transferSizes;
         this.eventMetadata = eventMetadata;
         this.mapper = mapper;
         this.deleteFilesOnPath = deleteFilesOnPath;
+        this.metricRegistry = metricRegistry;
+        this.metricConfig = metricConfig;
     }
 
     @Override
     public DownloadJob call()
+    {
+        if ( metricConfig == null || !metricConfig.isEnabled() || metricRegistry == null )
+        {
+            return doCall();
+        }
+
+        logger.trace( "Download metric enabled, location: {}", location );
+
+        Timer repoTimer = null;
```

Comment:
Can we create a generic second timer for all http downloads and use that here too? That way, we have a way to sort of aggregate / average all upstream calls for everything including the ones we don't single out, and can to 99th-percentile graphing of that time to look for a repo that might be dragging down performance.

Also, can we get call counts as well as timing info?

Reply:
A globalTimer is a few lines below for all downloads. :)
According to https://stackoverflow.com/questions/30987757/dropwizard-metrics-meters-vs-timers, a timer also provides "a count of the number of events".